sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoByteBuddyUtils.java

```diff
@@ -190,15 +191,59 @@ class ProtoByteBuddyUtils {
   private static final String DEFAULT_PROTO_GETTER_PREFIX = "get";
   private static final String DEFAULT_PROTO_SETTER_PREFIX = "set";
 
+  // BEAM-14525: there is a slight difference between 'protoc' and Guava CaseFormat regarding the
+  // camel case conversion
+  // - guava keeps the first character after a number lower case
+  // - protoc makes it upper case
+  // based on
+  // https://github.com/protocolbuffers/protobuf/blob/ec79d0d328c7e6cea15cc27fbeb9b018ca289590/src/google/protobuf/compiler/java/helpers.cc#L173-L208
+  @VisibleForTesting
+  static String convertProtoPropertyNameToJavaPropertyName(String input) {
```

Comment:
```suggestion
  static String underscoreToUpperCamel(String input) {
```

Reply:
I originally went a name similar to this, but decided in favor of another for multiple reasons:
- there seems to be no standard regarding conversion between various case formats - therefore this name isn't necessarily accurate -
- it's not even guaranteed that the proto property is in lower underscore (foo2bar should be foo_2_bar IMO)
- it's also not determined if foo2bar should be Foo2Bar or Foo2bar, there are converters for both cases

I prefer to name methods as accurately as possible according to their behaviour and not regarding an implementation detail. The method name I proposed is accurate. The fact that the conversion (almost looks like or) is a lower underscore to upper camel case is an implementation detail and irrelevant. Also as it's not a standard can't be 100% accurate. Especially that ignores any special (non-alphanumeric) characters I really doubt thats how that conversion should work - so IMO it would be a bad idea for anyone to call it who needs a lower underscore to upper camel conversion.

Reply:
@kileys

Reply:
@aaltay @kileys I'm kinda lost now. If this is going to be a discussion, then turns taking weeks is going make it quite a long one for a bugfix.

Reply:
@aaltay @kileys I'm kinda lost now. If this is going to be a discussion, then turns taking weeks is going make it quite a long one for a bugfix.